selfie.c

```diff
@@ -5488,6 +5487,7 @@ void emit_bootstrapping() {
     // avoid sign extension that would result in an additional sub instruction
     if (gp_value < two_to_the_power_of(31) - two_to_the_power_of(11))
       // assert: generates no more than two instructions
+      // assert: generates no data in data segment
```

Comment:
Please explain this comment. I am not sure I understand.

Reply:
I documented a bit more verbose in the updated commit.

Essentially the problem comes from the fact, that `load_integer()` does allocate memory on the data segment for big values. But at this point, there are no data segment allocations allowed, since the `gp_value` is already calculated. 

Even tough we do check for `gp_value` here, it is not clear, that we do that for 2 reasons:
 - maximum of `n` amount of emitted instructions in `load_integer()`
 - `load_integer()` does not allocate memory in the data segment

Reply:
I also thought of directly emitting the necessary instructions in place, instead of calling `load_integer()`, because it relies on some implementation details of `load_integer()`, which is probably not good.